### ISSUE TITLE
Memory error with tensorflow GPU

### ISSUE THREAD

**Issue by pathakrohit08** [tensorflow/tensorflow#21500]
I have NVIDIA GPU Titan V installed on my remote machine and I am trying to do simple multi class lassification using CNN

`train_file = './data/csv_data.csv.zip'`
 `x_raw, y_raw, df, labels = data_helper.load_data_and_labels(train_file)`

here my csv_data zip file is 65MB .

`max_document_length = max([len(x.split(' ')) for x in x_raw])`
`logging.info('The maximum length of all sentences: {}'.format(max_document_length))`
`vocab_processor = learn.preprocessing.VocabularyProcessor(max_document_length)`
`x = np.array(list(vocab_processor.fit_transform(x_raw)))`
`y = np.array(y_raw)`

I am getting memory error at 
`vocab_processor = learn.preprocessing.VocabularyProcessor(max_document_length)`

I think it is trying to load all the data in memory which is obviously a bad practice. How can I optimize this code to load it in the memory efficiently.

Can anyone please point me to a detailed example or tutorial? 

Thanks

**Comment by pathakrohit08** [comment]
Have I written custom code  **Yes**
OS Platform and Distribution  **Ubuntu 16.04**
TensorFlow installed from   **https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow_gpu-1.10.0-cp27-none-linux_x86_64.whl**

TensorFlow version  **1.9**
Bazel version 
CUDA/cuDNN version   **i think 7**
GPU model and memory **NVIDIA Titan V 12288MB**
Exact command to reproduce **https://github.com/pathakrohit08/EatItRepository/blob/master/tf.py**
Mobile device  **No**

**Comment by asimshankar** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
